hashintel/hash
# Implement new entity editor

## 🌟 What is the purpose of this PR?

This PR introduces the new entity editor with the new designs, which has more features like showing links, providing a table for editing properties etc.

There will be follow-up PRs to introduce more functionality. Going to open multiple PRs as follow-ups for better reviewing experience.

⚠️ Designs except the property table should be accurate. (consider the Notion doc below, we left some design elements for later iterations)

## 🔗 Related links

[Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=4864%3A241900) _(internal)_
[What we'll leave for the first version of the Entity Editor](https://www.notion.so/hashintel/Entity-Editor-design-review-a47e537ce3af43b3b096ceaacdbe60d9) _(internal)_

## 🔍 What does this change?

## 🐾 Next steps
- [ ] Move shared components e.g. `OntologyChip` & `WhiteCard` to shared folder 
- [ ] Split `property-table.tsx` into smaller chunks
- [ ] implement snackbar.error(`Failed to update ${propertyTitle}`)
- [ ] Add a default sorting to properties table
- [ ] Add sorting on properties table
- [ ] Render tooltips on the table value column
- [ ] Links section
- [ ] Show “empty values” chip on properties section header
- [ ] Correct links on breadcrumbs (needs `/entities` & `/types` pages)
- [ ] Improve loading state
- [ ] consider using `extractBaseUri` function for entity type href, when the need to call `init` gets resolved
- [ ] Property table type support
    - [ ] array
    - [ ] object


## ❓ How to test this?

1.  Checkout the branch / view the deployment
2. Login
3. Open `/entity-editor` url, and copy the `entityId` of logged in user there. It should be sth like `a0718f14-5806-458b-825a-cc00f8b92b8f`
4. Open `/@username/entities/{entityId}` url by using the `entityId` you just copied


## 📹 Demo
![Screen Shot 2022-10-12 at 15 22 44](https://user-images.githubusercontent.com/39553853/195341526-261816a0-9c88-4af6-a18c-45e784a3d0c2.png)
![Screen Shot 2022-10-12 at 15 23 10](https://user-images.githubusercontent.com/39553853/195341609-66ac0f7d-c717-4e16-a0ee-a6f14289dd40.png)


https://user-images.githubusercontent.com/39553853/195341426-09e13da5-d576-4988-b774-75c3a1e89479.mov
